extractor/main.go

```diff
@@ -0,0 +1,307 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/gnolang/gno/gno.land/pkg/sdk/vm"
+	"github.com/gnolang/gno/tm2/pkg/amino"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/peterbourgon/ff/v3/ffcli"
+	"golang.org/x/sync/errgroup"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// Define constants
+const (
+	packageMetadataFile = "pkg_metadata.json"
+)
+
+// Define errors
+var (
+	errInvalidFileType  = errors.New("invalid file type specified")
+	errInvalidSourceDir = errors.New("invalid source directory")
+	errInvalidOutputDir = errors.New("invalid output directory")
+)
```

Comment:
these you can just have as errors returned by `execExtract`

defining error types as global variables is useful if the errors need to be checked (ie. `errors.Is(err, errInvalidFileType)`). it's not the case here, also because you are writing an executable, not a library